lddwrap/__init__.py

```diff
@@ -0,0 +1,228 @@
+    if not path.is_file():
+        raise IOError("Not a file: {}".format(path))
+
+    if verbose and unused:
+        get_dependencies(path=path, verbose=True)
+        verbose = False
+
+    if verbose:
+        commands.append("--verbose")
+
+    commands.append(path.as_posix())
+
+    proc = subprocess.Popen(commands, stdout=subprocess.PIPE, stderr=subprocess.PIPE, universal_newlines=True, env=env)
+
+    out, err = proc.communicate()
+    if proc.returncode != 0:
+        raise RuntimeError("Failed ldd-ing external libraries of {} with code {}:\nout:\n{}\n\nerr:\n{}".format(
+            path, proc.returncode, out, err))
+
+    out_unused = ""
+    if unused:
+        commands = ["ldd", "--unused", path.as_posix()]
+        proc_unused = subprocess.Popen(
+            commands, stdout=subprocess.PIPE, stderr=subprocess.PIPE, universal_newlines=True, env=env)
+        out_unused, err_unused = proc_unused.communicate()
+        # return code = 0 -> no unused dependencies, return code = 1 -> some unused dependencies
+        if proc_unused.returncode not in [0, 1]:
+            raise RuntimeError("Failed ldd-ing external libraries of {} with code {}:\nout:\n{}\n\nerr:\n{}".format(
+                path, proc.returncode, out_unused, err_unused))
+
+    dependencies = dependencies_parse_helper(
+        out=out, out_unused=out_unused, verbose=verbose, unused=unused)  # type: List[Dependency]
+
+    return dependencies
+
+
+def dependencies_parse_helper(out: str, out_unused: str, verbose: Optional[bool] = None,
+                              unused: Optional[bool] = None) -> List[Dependency]:
+    """
+    Help to parse the command line output.
+
+    :param out: command line output
+    :param out_unused: command line output for ldd --unused
+    :param verbose: verbose flag set
+    :param unused: unused flag set
+    """
+    if verbose:
+        print(out)
+
+    dependencies = []  # type: List[Dependency]
+    for line in [line for line in out.split('\n') if line.strip()]:
+        # end of output
+        if "Version information" not in line:
+            dep = parse_line(line=line)
+            if dep is not None:
+                dependencies.append(dep)
+        else:
+            break
+
+    if unused:
+        dependencies = set_unused(dependencies=dependencies, out_unused=out_unused)
+
+    return dependencies
+
+
+def set_unused(dependencies: List[Dependency], out_unused: str) -> List[Dependency]:
+    """
+    Set Dependency argument unused.
+
+    :param dependencies: List of dependencies
+    :param out_unused: output from command ldd --unused
+    :return: List of dependencies
+    """
+    unused_dependencies = []  # type: List[pathlib.Path]
+    for line in [line for line in out_unused.split('\n') if line.strip()]:
+        # skip first line because it's no dependency
+        if line != "Unused direct dependencies:":
+            unused_dependencies.append(pathlib.Path(line.strip()))
+
```

Comment:
I think these nested loops are buggy. You should set `dep.unused = False` only once for each dependency. The way the code is written now could result in `False` overwriting a `True`. I would suggest writing this code like this:

```
    for dep in dependencies:
        dep.unused = False
        for unused_dep in unused_dependencies:
            if unused_dep == dep.path:
                dep.unused = True
                break;
```

Please let me know if you disagree :)

Reply:
I think the logic behind isn't buggy but I do some unnecessary loops. Your version is cleaner.